### ISSUE TITLE
Evaluating best model for script classification.

### ISSUE THREAD

**Issue by karmatai** [OpenPecha/BEC#318]
# Description
In this card we are going to  evaluate which model works best for the current script classifying task and publish the result.

**Repo:** https://github.com/OpenPecha/script-classification-model-train
**Doc:** https://docs.google.com/document/d/1HYaujDST_A31DIbByDcrXqzdstuoSUbJb0xMbBA2zIY/edit?tab=t.0
## Models 
- [x] DINOv2 ViT-S/14 	
- [x] DINOv2 ViT-B/14 	
- [x] EfficientNet-B0 	
- [x] CLIP ViT-B/32 	
- [x] ConvNeXt-Base 	
- [x] ResNet-50 	
- [x] DINOv3 ViT-S
- [x] SigLIP 2 Base

## Subtask
- [x] Prepare the dataset for linear probe shootout.
- [x] Implement Linear probe shootout for each model.
- [x] Upload data to hugging face.
- [x] Analys the confusion matrix and pick top two.

## Linear probe result 
This report summarizes the results of a "Linear Probe Shootout" conducted to identify the most effective pre-trained model (backbone) for classifying Tibetan scripts. We tested 8 architectures using 1,467 annotated images across 22 script classes. The models were frozen, and a simple Logistic Regression was used to evaluate the quality of the extracted embeddings.

|   index | Model            |   Embedding Dim |   Extract Time (s) |   Accuracy |   Macro F1 |   Weighted F1 |
|--------:|:-----------------|----------------:|-------------------:|-----------:|-----------:|--------------:|
|       0 | DINOv2 ViT-S/14  |             384 |             196.28 |     0.4431 |     0.3846 |        0.4438 |
|       1 | DINOv2 ViT-B/14  |             768 |             308.38 |     0.4567 |     0.3923 |        0.4572 |
|       2 | EfficientNet-B0  |            1280 |             112.82 |     0.4213 |     0.3532 |        0.4136 |
|       3 | CLIP ViT-B/32    |             512 |             111.35 |     0.3926 |     0.283  |        0.348  |
|       4 | ConvNeXt-Base    |            1024 |             133.21 |     0.4322 |     0.3739 |        0.431  |
|       5 | ResNet-50 (~25M) |            2048 |             116    |     0.4172 |     0.3472 |        0.407  |
|       6 | DINOv3 ViT-S |            384 |           125.14 |          0.4956 |          0.4351 |          0.49393 |
|       7 | SigLIP 2 Base |          768 |         132.80 |          0.4717 |           0.4177 |          0.46995 |
         
### 1. The Performance Leader: DINOv2
**DINOv3 ViT-S** achieved the highest performance across all primary metrics, including a **Macro F1 of 0.4351**. This indicates that DINOv2 provides the most discriminative features for Tibetan script morphology, even without fine-tuning.

### 2. The Efficiency Trade-off
* **Fastest:** CLIP ViT-B/32 and EfficientNet-B0 are significantly faster, processing the dataset in ~112 seconds.
* **Smartest:** DINOv3 ViT-S has the highest accuracy followed by **SigLIP 2 base.**
* **Best Choice:** **DINOv3 ViT-S** current shows the best accuracy on an median extract time.

### Confusion matrix for DINOv2 ViT-S/14

<img width="866" height="790" alt="Image" src="https://github.com/user-attachments/assets/6b089623-b22e-4656-b432-70992a803628" />

<img width="866" height="790" alt="Image" src="https://github.com/user-attachments/assets/cf4f6ca5-39f2-449f-8e70-e45aab7753e8" />

## Reviewer
- [x] @gangagyatso4364

**Comment by 10zYmon** [comment]
DINOv3 ViT-S gives best results, working on script to fine tune DINOv3 ViT-S.
